src/kernel/mbrowse.ml

```diff
@@ -261,3 +277,39 @@ let print_node () node =
 
 let print () t =
   List.print (fun () (_,node) -> print_node () node) () t
+
+
+let check_node pos node =
+  let loc = node_merlin_loc node in 
+  if Location_aux.compare_pos pos loc = 0 then true else false 
+
+let get_children pos root = 
+  let children = 
+    match root with
+    | (_,Structure str) -> 
+        of_structure_items (List.filter ~f:(fun (_,x) ->
+            check_node pos (Structure_item (x, str.str_final_env))
+        ) (List.map ~f:(fun x -> (str.str_final_env, x)) str.str_items))
+    | (_,Signature str) -> 
+      of_signature_items (List.filter ~f:(fun (_,x) ->
+        check_node pos (Signature_item (x, str.sig_final_env))
+      ) (List.map ~f:(fun x -> (str.sig_final_env, x)) str.sig_items))
```

Comment:
I guess that we can simplify `get_children` using `filter_map` or moving some logic in `of_(signature|structure)_items`. WDYT?